RecoEgamma/ElectronIdentification/src/ElectronMVAEstimatorRun2Fall17.cc

```diff
@@ -1,472 +0,0 @@
-  auto superCluster = eleRecoPtr->superCluster();
-
-  // Pure ECAL -> shower shapes
-  float see            = eleRecoPtr->full5x5_sigmaIetaIeta();
-  float spp            = eleRecoPtr->full5x5_sigmaIphiIphi();
-  float circularity = 1. - eleRecoPtr->full5x5_e1x5() / eleRecoPtr->full5x5_e5x5();
-  float r9             = eleRecoPtr->full5x5_r9();
-  float etawidth       = superCluster->etaWidth();
-  float phiwidth       = superCluster->phiWidth();
-  float hoe            = eleRecoPtr->full5x5_hcalOverEcal(); //hadronicOverEm();
-  // Endcap only variables
-  float preShowerOverRaw  = superCluster->preshowerEnergy() / superCluster->rawEnergy();
-
-  // To get to CTF track information in pat::Electron, we have to have the pointer
-  // to pat::Electron, it is not accessible from the pointer to reco::GsfElectron.
-  // This behavior is reported and is expected to change in the future (post-7.4.5 some time).
-  bool validKF= false;
-  reco::TrackRef myTrackRef = eleRecoPtr->closestCtfTrackRef();
-  const pat::Electron * elePatPtr = dynamic_cast<const pat::Electron *>(eleRecoPtr);
-  // Check if this is really a pat::Electron, and if yes, get the track ref from this new
-  // pointer instead
-  if( elePatPtr != nullptr ) {
-    myTrackRef = elePatPtr->closestCtfTrackRef();
-  }
-  validKF = (myTrackRef.isAvailable() && (myTrackRef.isNonnull()) );
-
-  //Pure tracking variables
-  float kfhits         = (validKF) ? myTrackRef->hitPattern().trackerLayersWithMeasurement() : -1. ;
-  float kfchi2          = (validKF) ? myTrackRef->normalizedChi2() : 0;
-  float gsfchi2         = eleRecoPtr->gsfTrack()->normalizedChi2();
-
-  // Energy matching
-  float fbrem           = eleRecoPtr->fbrem();
-
-  float gsfhits         = eleRecoPtr->gsfTrack()->hitPattern().trackerLayersWithMeasurement();
-  float expectedMissingInnerHits = eleRecoPtr->gsfTrack()
-    ->hitPattern().numberOfLostHits(reco::HitPattern::MISSING_INNER_HITS);
-
-  reco::ConversionRef convRef = ConversionTools::matchedConversion(*eleRecoPtr,
-                                    conversions,
-                                    theBeamSpot->position());
-  float convVtxFitProbability = -1.;
-  if(!convRef.isNull()) {
-    const reco::Vertex &vtx = convRef.get()->conversionVertex();
-    if (vtx.isValid()) {
-      convVtxFitProbability = (float)TMath::Prob( vtx.chi2(), vtx.ndof());
-    }
-  }
-
-  float eop             = eleRecoPtr->eSuperClusterOverP();
-  float eleeopout       = eleRecoPtr->eEleClusterOverPout();
-  float pAtVertex            = eleRecoPtr->trackMomentumAtVtx().R();
-  float oneOverEminusOneOverP         = (1.0/eleRecoPtr->ecalEnergy()) - (1.0 / pAtVertex );
-
-  // Geometrical matchings
-  float deta            = eleRecoPtr->deltaEtaSuperClusterTrackAtVtx();
-  float dphi            = eleRecoPtr->deltaPhiSuperClusterTrackAtVtx();
-  float detacalo        = eleRecoPtr->deltaEtaSeedClusterTrackAtCalo();
```

Comment:
Hello! So we saw some very tiny non-agreement for the Fall17 V1 ID, which I investigated:

https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_10_2_X_2018-06-06-2300+23473/26978/1325.7_TTbar_13_94Xv2NanoAODINPUT+TTbar_13_94Xv2NanoAODINPUT+NANOEDMMC2017_94XMiniAODv2+HARVESTNANOAODMC2017_94XMiniAODv2/Physics_NanoAODDQM_Electron.html

This is related to the fact that instead of taking the raw geometrical track-cluster distances, this PR now takes the absolute value of them, to be consistent with the old ID:

https://github.com/guitargeek/cmssw/blob/1ebc151698c13f38367614b41f6a1cf1fd16f8b8/RecoEgamma/ElectronIdentification/data/ElectronMVAEstimatorRun2Variables.txt#L36

If we care about the precise numerical agreement here, I could also change this by using a separate variable text file for Fall17 V1 where the absolute values are not taken. @Sam-Harper, @lsoffi, @michelif: any opinion on this? I guess you have plenty of feedback besides this, @perrotta and @slava77, so I can tackle this nuisance together with your other comments once available.

By the way, I also made some comparison of all MVA outputs myself, before and after the PR (for AOD and MiniAOD):

https://github.com/guitargeek/cmssw/blob/1ebc151698c13f38367614b41f6a1cf1fd16f8b8/RecoEgamma/ElectronIdentification/data/ElectronMVAEstimatorRun2Variables.txt

You can see the same we see here: numerical agreement for all IDs besides a very tiny difference in Fall17 V1.

Reply:
@guitargeek To me, it would be nice if we could exactly reproduce Fall17V1 in this release if its reasonably feasible and not too much maintenance.